### PR TITLE
Feature/284

### PR DESCRIPTION
**Pull request by ki-arie**
Following deets in this [ticket](https://github.com/TeHikuMedia/waha-tuhi/issues/284)

### CODE REVIEW

File: native_client/deepspeech.cc [hunk 1]
```diff
@@ -239,6 +250,24 @@ StreamingState::processMfccWindow(const vector<float>& buf)
   }
 }
 
+vector<double> StreamingState::flatten(xarray<double> logits){
```
Comment (utunga) [inline native_client/deepspeech.cc:253]: tiny thing but should the variables be called logits/rawLogits etc given that these functions are basically generic?

File: native_client/deepspeech.cc [hunk 2]
```diff
@@ -254,8 +283,15 @@ StreamingState::processBatch(const vector<float>& buf, unsigned int n_steps)
...
-  vector<double> inputs(logits.begin(), logits.end());
+  // 1. Reshape rolled out logits
+  xarray<double> reshapedRawLogits = reshape(logits, num_classes, n_frames);
+
+  // 2. Perform softmax
+  xarray<double> softmaxLogits = softmax(reshapedRawLogits);
+
+  // 3. Rollout softmax logits
+  vector<double> inputs = flatten(softmaxLogits);
+
```
Comment (utunga) [inline native_client/deepspeech.cc:294]: 👍

Comment (utunga) [reply native_client/deepspeech.cc:294]: Why do you think they were converting  floats to double at this point ? probably a good idea for our context though anyway

**Review by utunga — APPROVED** [review]
LGTM.

To be honest the tricky things are (a) building and (b) verifying that this gives correct behavior, but both those changes are out of scope of these tweaks
